improver/orographic_enhancement.py

```diff
@@ -36,6 +36,106 @@
+                enhancement, as proxy for the boundary layer.
+            boundary_height_units (str):
+                Units of the boundary height specified for extracting model levels.
+
+        """
+        self._constraint_info = (boundary_height, boundary_height_units)
+
+    @staticmethod
+    def extract_and_check(cube, height_value, units):
+        """
+        Function to attempt to extract a height level.
+        If no matching level is available an error is raised.
+
+        Args:
+            cube (cube):
+                Cube to be extracted from and checked it worked.
+            height_value (float):
+                The boundary height to be extracted with the input units.
+            units (str):
+                The units of the height level to be extracted.
+
+        Returns:
+            iris.cube.Cube:
+                A cube containing the extracted height level.
+
+        Raises:
+            ValueError: If height level is not found in the input cube.
+
+        """
+        # Write constraint in this format so a constraint is constructed that
+        # is suitable for floating point comparison
+        height_constraint = [
+            "height=[{}:{}]".format(height_value - 0.1, height_value + 0.1)
+        ]
+        cube = extract_subcube(cube, height_constraint, units=[units])
+
+        if cube is not None:
+            return cube
+
+        raise ValueError("No data available at height {}{}".format(height_value, units))
+
+    def process(self, *cubes: Cube | CubeList) -> Cube:
+        """
+        Uses the ResolveWindComponents() and OrographicEnhancement() plugins.
+        Outputs data on the high resolution orography grid.
+
+        Args:
+            cubes:
+                temperature:
+                    Cube containing temperature at top of boundary layer.
+                humidity:
+                    Cube containing relative humidity at top of boundary layer.
+                pressure:
+                    Cube containing pressure at top of boundary layer.
+                wind_speed:
+                    Cube containing wind speed values.
+                wind_direction:
+                    Cube containing wind direction values relative to true north.
+                orography:
+                    Cube containing height of orography above sea level on high
+                    resolution (1 km) UKPP domain grid.
+        Returns:
+            iris.cube.Cube:
+                Precipitation enhancement due to orography on the high resolution
+                input orography grid.
+
+        """
+        from improver.wind_calculations.wind_components import ResolveWindComponents
+
+        cubes = as_cubelist(*cubes)
+        for ind in range(len(cubes)):
+            if "surface_altitude" in cubes[ind].name():
+                continue
+            cubes[ind] = self.extract_and_check(cubes[ind], *self._constraint_info)
+
```

Comment:
I derived these names via the acceptance testing.  Based on my previous experience for other acceptance tests, sometimes the acceptance test data has had incorrect names.  Please verify the validity of the names used here for the extraction.